src/carousel.js

```diff
@@ -57,9 +53,28 @@ class Carousel extends React.Component {
     this.refs.wrapper.removeEventListener('touchend', this.onTouchEnd, {capture: true})
   }
 
-  componentWillReceiveProps(nextProps) {
+  componentDidUpdate(_, prevState) {
+    if (this.state.frames.length && this.state.frames.length !== prevState.frames.length) {
+      // reset to default state
+      this.hideFrames()
+      translateXY(this.refs['f0'], 0, 0, 0)
+    }
+  }
+
+  static getDerivedStateFromProps(nextProps, prevState) {
     const frames = [].concat(nextProps.frames || nextProps.children || [])
-    this.setState({ frames })
+    const nextState = { frames } 
+    if (frames.length && frames.length !== prevState.frames.length) {
+      nextState.current = 0
+    }
+   
```

Comment:
Extra spaces

Reply:
got it 👍

Reply:
@amio  this has been addressed in latest commit.